optimum/habana/transformers/models/gemma/modeling_gemma.py

```diff
@@ -652,7 +652,7 @@ def forward(
             position_ids = position_ids.unsqueeze(0)
 
         # HPU specific mask generation
-        if attention_mask is not None and attention_mask.dim() != 4:
+        if attention_mask is None or attention_mask.dim() != 4:
             attention_mask = _gaudi_prepare_4d_causal_attention_mask(
```

Comment:
if attention_mask is none, why do we still use this?

Reply:
It creates a causal mask when attention_mask is None. The updated criteria is equivalent to https://github.com/huggingface/transformers/blob/53fad641cfdb5105e2470bcf3ef17ea8e25cc300/src/transformers/models/gemma/modeling_gemma.py#L532 

```
if attention_mask is not None and attention_mask.dim() == 4:
        # In this case we assume that the mask comes already in inverted form and requires no inversion or slicing.
        causal_mask = attention_mask
```